R/NielsonKuparuk.R

```diff
@@ -0,0 +1,278 @@
+##Neilson, 2017
+##Megan Nguyen
+##2018
+
+#Clean files
+IC_WeirQ <- read.csv("/home/mnguyen/Neilson/IC_WeirQ.csv", header = TRUE)
+IC_WeirQ <- IC_WeirQ[,-3]
+names(IC_WeirQ) <- c("Date_adt", "Discharge_cms")
+
+Kup2AQ <- read.csv("/home/mnguyen/Neilson/Kup2AQ.csv", header = TRUE)
+names(Kup2AQ)
+
+ICKUP_Q <- read.csv("/home/mnguyen/Neilson/ICKUP_Q.csv", header = TRUE)
+str(ICKUP_Q)
+results <- grepl(pattern = "^\\d", ICKUP_Q$Date_adt)
+delete <- c()
+for(i in 1:length(ICKUP_Q$Date_adt)){
+  if(results[i] == FALSE){
+    delete <- c(delete, i)
+  }
+}
+
+ICKUP_Q <- ICKUP_Q[-delete,]
+names(ICKUP_Q)
+
+Kup2TQ <- read.csv("/home/mnguyen/Neilson/Kup2TQ.csv", header = TRUE)
+
+Kup3BQ <- read.csv("/home/mnguyen/Neilson/Kup3BQ.csv", header = TRUE)
+
+Kup3TQ <- read.csv("/home/mnguyen/Neilson/Kup3TQ.csv", header = TRUE)
+
+Kup4Q <- read.csv("/home/mnguyen/Neilson/Kup4Q.csv", header = TRUE)
+
+Kup5Q <- read.csv("/home/mnguyen/Neilson/Kup5Q.csv", header = TRUE)
+names(Kup5Q) <- c("Date_adt", "Discharge_cms")
+
+Kup8Q <- read.csv("/home/mnguyen/Neilson/Kup8Q.csv", header = TRUE)
+names(Kup8Q) <- c("Date_adt", "Discharge_cms")
+
+Kup9Q <- read.csv("/home/mnguyen/Neilson/Kup9Q.csv", header = TRUE)
+results <- grepl(pattern = "^\\d", Kup9Q$Date_adt)
+delete <- c()
+for(i in 1:length(Kup9Q$Date_adt)){
+  if(results[i] == FALSE){
+    delete <- c(delete, i)
+  }
+}
+
+Kup9Q <- Kup9Q[-delete,]
+
+
+files <- list(IC_WeirQ, ICKUP_Q, Kup2AQ, Kup2TQ, Kup3BQ, Kup3TQ, Kup4Q, Kup5Q, Kup8Q, Kup9Q)
+
+#Separate date and time in order to edit
+
+for(i in 1:length(files)){
+  files[[i]] <- separate(data = files[[i]], col = Date_adt, into = c("Date", "Time"), sep = " ")
+}
+
+##Add ":00" at end of time
+for(i in 1:length(files)){
+  for(j in 1:length(files[[i]]$Time)){
+    files[[i]]$Time[j] <- paste(files[[i]]$Time[j], ":00", sep = "")
```

Comment:
This inner loop looks like a prime example of a step you can try using the tidyverse, which aims to make more readable, intuitive code.  See here: http://dplyr.tidyverse.org/

In this case, the tidyverse version of your code would be something along these lines:

```
library(tidyverse)
files[[i]] <- files[[i]] %>% 
  mutate(Time = paste0(Time, ":00"))
```

If you want to stick with base R, you can also take advantage of R's vectorization abilities to avoid the inner for loop:  (let me know if this doesn't work)

```
files[[i]]$Time <- paste0(files[[i]]$Time, ":00")
```

Reply:
also note: `paste0` is the same as `paste` except that sep = ""